xarray/backends/netCDF4_.py

```diff
@@ -210,9 +216,9 @@ def __init__(self, filename, mode='r', format='NETCDF4', group=None,
         self.ds = opener()
```

Comment:
In theory, I think we could support pickling if passed all the right keyword arguments in addition to the existing `netCDF4.Dataset` object, and we used the filename/group from the input argument as arguments to `opener`. But I'm not entirely sure that's worth doing.

It's probably simpler for now to explicitly raise an error for cases where we would re-use an opener, namely when pickling and/or using `autoclose=True`.  Can you add that? Right now I would guess that using either of those options could result in somewhat opaque errors.